mtags/src/main/scala-3/scala/meta/internal/pc/completions/OverrideCompletions.scala

```diff
@@ -104,6 +113,150 @@ object OverrideCompletions:
       .toList
   end contribute
 
+  def implementAllFor(
+      indexedContext: IndexedContext,
+      search: SymbolSearch,
+      autoImports: AutoImportsGenerator,
+      config: PresentationCompilerConfig
+  )(
+      td: TypeDef
+  )(using Context): List[l.TextEdit] =
+    def calcIndent(
+        td: TypeDef,
+        decls: List[Symbol],
+        source: SourceFile,
+        text: String,
+        shouldCompleteBraces: Boolean
+    )(using Context): (String, String, String) =
+      // For `FooImpl` in the below, the necessaryIndent will be 2
+      // because there're 2 spaces before `class FooImpl`.
+      // ```scala
+      // |object X:
+      // |  class FooImpl extends Foo {
+      // |  }
+      // ```
+      val necessaryIndent = inferIndent(
+        source.lineToOffset(td.sourcePos.line),
+        text
+      )
+      // infer indent for implementations
+      // If there's declaration in the class/object, follow its indent.
+      // For example, numIndent will be 8, becuase there're 8 spaces before
+      // `override def foo: Int`
+      // ```scala
+      // |object X:
+      // |  class FooImpl extends Foo {
+      // |        override def foo: Int = 1
+      // |  }
+      // ```
+      val numIndent =
+        decls.headOption
+          .map { decl =>
+            inferIndent(source.lineToOffset(decl.sourcePos.line), text)
```

Comment:
Let's make sure that this works with tabs

Reply:
Can I work on support tabs in separate issues / PR? It's pretty complex than I expected to support tabs.
(first of all, should we support tab? Scala2 version neither support tab, but I'd never seen the report to support tabs for override completion / implement all code action for Scala2)

Here I tried to support tabs https://github.com/tanishiking/metals/pull/16/commits/ddca731f1be8bc88b0a20a7ee250497099c457f0 but it doesn't work, for example, `` returns


```scala
    """|package a
       |
       |object A {
       |        trait Base {
       |                def foo(x: Int): Int
       |                def bar(x: String): String
       |        }
       |
       |        class Concrete extends Base 
       |
       |                override def foo(x: Int): Int = ???
       |
       |                override def bar(x: String): String = ???
       |
       |        {}
       |
       |}
       |""".stripMargin
```

Looks like this is because Metals / Scala compiler counts `\t` as 1 character, while the LSP client (and test?) counts `\t` as 2 characters (maybe?) 🤔

Reply:
I think in a lot of places it's automatically converted to spaces, so using tab will just do that.

Reply:
There was related PR https://github.com/scalameta/metals/pull/3702/files but since you are not using `offsetToLine` it should fine.

Reply:
Och, but if someone has tabs we should use that instead of " " anyway, but it's fine to do a follow up.

Reply:
https://github.com/scalameta/metals/issues/4027 Opened an issue :+1:

Reply:
Have you seen the `getIndentForPos` method on the flatMap PR?